external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -2851,103 +3152,124 @@ pub fn on_use_request(
 
 /// Handles document symbol request of the language server
 #[allow(deprecated)]
-pub fn on_document_symbol_request(context: &Context, request: &Request, symbols: &Symbols) {
+pub fn on_document_symbol_request(context: &Context, request: &Request) {
+    let symbols_map = &context.symbols.lock().unwrap();
     let parameters = serde_json::from_value::<DocumentSymbolParams>(request.params.clone())
         .expect("could not deserialize document symbol request");
 
     let fpath = parameters.text_document.uri.to_file_path().unwrap();
     eprintln!("on_document_symbol_request: {:?}", fpath);
 
-    let empty_mods: BTreeSet<ModuleDefs> = BTreeSet::new();
-    let mods = symbols.file_mods.get(&fpath).unwrap_or(&empty_mods);
-
     let mut defs: Vec<DocumentSymbol> = vec![];
-    for mod_def in mods {
-        let name = mod_def.ident.module.clone().to_string();
-        let detail = Some(mod_def.ident.clone().to_string());
-        let kind = SymbolKind::MODULE;
-        let Some(range) = symbols.files.lsp_range_opt(&mod_def.name_loc) else {
-            continue;
-        };
-
-        let mut children = vec![];
-
-        // handle constants
-        let cloned_const_def = mod_def.constants.clone();
```

Comment:
Cloning here and other similar places was unnecessary so I fixed it while I was modifying these functions to support enums.